### PR TITLE
Add options for personal schedule and favourite sessions for ticket holders

### PR DESCRIPTION
**Pull request by odkhang**
This PR closes/references issue #136 . It does so by:
- Implement save favorite talk to database and showing on schedule.
- #141 to show do not record on schedule view.
- Call to video component to sync favorite talk

## Checklist

- [ ] I have added tests to cover my changes.

## Summary by Sourcery

This pull request adds functionality for users to save and manage their favorite talks, integrating this feature into the personal schedule. It also enhances the schedule view to show additional information about talks, such as the number of favorites and recording preferences.

- **New Features**:
    - Introduced the ability for users to save their favorite talks to a personal schedule.
    - Added a new API endpoint to handle user's favorite talks, allowing retrieval and addition of favorite talks.
- **Enhancements**:
    - Updated the schedule view to display the count of favorites for each talk and indicate if a talk should not be recorded.

### CODE REVIEW

**Comment by odkhang** [comment]
image

With this PR:
- Add option to sort by Time/Title/Popularity (sort menu layout not work with css from schedule plugin yet)
- Show number of user favorite a session.
- Show do not record on session

video PR: https://github.com/fossasia/eventyay-video/pull/176

**Comment by mariobehling** [comment]
1. Please follow the sourcery suggestions.
2. Use the same white or black font color depending on the background for the number circle (same as for the time and date).
3. Please explain how this can work for the video component. Is there an API connection?

**Comment by mariobehling** [comment]
Sorting does not work at all yet. I also see the option for different timezones. This is out of scope for this issue.

Screenshot from 2024-07-28 22-43-50

**Comment by odkhang** [comment]
Hi @mariobehling,

image

font color and  diameter of fav count is updated.
And the pop will look like above, do you have any feedback?

About the unstar issue, I open 2 browsers, if I unstar one, and refresh other brower, it also reflect the number, could you test again?

**Comment by odkhang** [comment]
About the connection with Video.

From Talk:
- when user star a session, if user existed in Video, call api to sync the list favorite sessions to Video for that user. (same when user un star a session)

From video:
 - when user star a session, trigger a API to Talk to update list favorite session in talk (same when user unstar a session)

File: src/pretalx/api/views/submission.py [hunk 1]
```diff
@@ -128,3 +145,118 @@
...
+        auth_header = get_authorization_header(request).split()
+        if not auth_header:
+            raise Http404
+        if auth_header and auth_header[0].lower() == b'bearer':
+            if len(auth_header) == 1:
+                raise exceptions.AuthenticationFailed('Invalid token header. No credentials provided.')
+            elif len(auth_header) > 2:
+                raise exceptions.AuthenticationFailed(
+                    'Invalid token header. Token string should not contain spaces.')
+        token_decode =  jwt.decode(auth_header[1], video_settings.secret,
```
[the remark this answers is not in the document]
Comment (odkhang) [reply src/pretalx/api/views/submission.py:256]: its not hard-coded, secret is getting from setting config

**Comment by mariobehling** [comment]
Please resolve conflicts.

**Pull request merged**